src/platform/screen.c

```diff
@@ -230,14 +230,33 @@ int system_scale_display(int display_scale_percentage)
     SDL_GetWindowSize(SDL.window, &width, &height);
     set_scale_percentage(display_scale_percentage, width, height);
     platform_screen_resize(width, height);
-    return scale_percentage;
+    return scale.percentage;
 }
 
-int system_get_max_display_scale(void)
+int system_can_scale_display(int *min_scale, int *max_scale)
 {
+#ifndef __ANDROID__
+    if (system_is_fullscreen_only()) {
+        return 0;
+    }
+#endif
     int width, height;
     SDL_GetWindowSize(SDL.window, &width, &height);
-    return get_max_scale_percentage(width, height);
+#ifdef __ANDROID__
+    int maxOrientation1 = get_max_scale_percentage(width, height);
```

Comment:
```suggestion
    int max_orientation_portrait = get_max_scale_percentage(width, height);
```

Reply:
Since we don't know whether the current orientation is portrait or landscape I opted for simply "1" and "2".

Reply:
Makes sense. What about `max_orientation_current` and `max_orientation_alternative` though?

Reply:
Good suggestion. I opted for `max_scale_current_orientation` and `max_scale_alternative_orientation` to make it clear we're dealing with a maximum scale.